### PR TITLE
WFLY-18389 <max-active-sessions/> causes sessions to expire prematurely using the HotRod-based HttpSession manager

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-18389

### CODE REVIEW

**Review by rhusar — APPROVED** [review]
👍🏼  LGTM

File: clustering/web/hotrod/src/main/java/org/wildfly/clustering/web/hotrod/session/HotRodSessionFactory.java [hunk 1]
```diff
@@ -54,17 +51,21 @@
...
 import org.wildfly.clustering.web.session.ImmutableSessionAttributes;
 import org.wildfly.clustering.web.session.ImmutableSessionMetaData;
 import org.wildfly.security.manager.WildFlySecurityManager;
 
 /**
+ * Factory for creating a {@link org.wildfly.clustering.web.session.Session} backed by a set of {@link RemoteCache} entries.
  * @author Paul Ferraro
+ * @param <C> the marshalling context type
+ * @param <V> the session attribute entry type
+ * @param <C> the local context type
```
Comment (rhusar) [inline clustering/web/hotrod/src/main/java/org/wildfly/clustering/web/hotrod/session/HotRodSessionFactory.java:66]: s/<C>/<L> - but we don't need to fix this now.

Comment (pferraro) [reply clustering/web/hotrod/src/main/java/org/wildfly/clustering/web/hotrod/session/HotRodSessionFactory.java]: Fixed.

**Comment by pferraro** [comment]
Added logic to expiration test case to reproduce issue (and detect future regressions).

**Pull request merged**